src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivitySource.cs

```diff
@@ -32,13 +33,13 @@ public ActivitySource(string name, string? version = "")
 
             if (s_allListeners.Count > 0)
             {
-                s_allListeners.EnumWithAction(listener => {
+                s_allListeners.EnumWithAction((listener, source) => {
                     var shouldListenTo = listener.ShouldListenTo;
-                    if (shouldListenTo != null && shouldListenTo(this))
+                    if (shouldListenTo != null && shouldListenTo((ActivitySource)source))
                     {
-                        AddListener(listener);
+                        ((ActivitySource)source).AddListener(listener);
```

Comment:
It'd be slightly better to avoid the double cast. And also avoid the `var` when the type isn't clear on the rhs. e.g.
```C#
s_allListeners.EnumWithAction((listener, source) =>
{
    Func<ActivitySource, bool>? shouldListenTo = listener.ShouldListenTo;
    if (shouldListenTo != null)
    {
        var activitySource = (ActivitySource)source;
        if (shouldListenTo(activitySource))
        {
            activitySource.AddListener(listener);
        }
    }
});
```

Reply:
done. thanks.